src/ui/Island/Realm.tsx

```diff
@@ -47,46 +51,28 @@ export default function Realm({
 
   const islandContext = useIslandContext()
   const groupRef = useRef<Konva.Group>(null)
-  const pathRef = useRef<Konva.Path>(null)
   const textRef = useRef<Konva.Text>(null)
-  const imageLayerRef = useRef<Konva.Image>(null)
-  const overlayRef = useRef<Konva.Path>(null)
   const partnerLogoRef = useRef<Konva.Image>(null)
 
+  const [pathRefs, addPathRef] = useMultiRef<Konva.Path>()
+  const [, addOverlayRefs] = useMultiRef<Konva.Path>()
+  const [, addImageLayerRef] = useMultiRef<Konva.Image>()
```

Comment:
actually why do we need to set these refs? I can't find a place where they are used after they are set. Is it internal Konva's thing?

Reply:
We don't, good catch